app/helpers/charting_helper.rb

```diff
@@ -15,24 +16,35 @@ def chart_remote(a_controller, options)
                             :height     => options[:height],
                             :rand       => "#{rand(999_999_999)}"),
                     options.slice(:id, :bgcolor, :width, :height))
+    elsif Charting.backend == :c3
+      c3chart_remote(url_for(:controller => a_controller,
+          :action => options[:action] || 'render_chart',
+        ),
+        options.slice(:id))
     end
   end
 
   def chart_no_url(options)
+    p :chart_no_url
     if Charting.backend == :ziya
       ziya_chart(nil, options.slice(:id, :bgcolor, :width, :height))
     elsif Charting.backend == :jqplot
       jqplot_sample(options.slice(:id, :bgcolor, :width, :height))
+    elsif Charting.backend == :c3
+      c3chart_sample
```

Comment:
Is it possible to use `case Charting.backend when ...` there? Also in `chart_remote` and `chart_local`.